scale/job/migrations/0016_auto_20160421_1553.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import models, migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('job', '0015_auto_20160420_1345'),
+        ('storage', '0002_workspace_is_move_enabled'),
+    ]
+
+    def populate_job_configuration(apps, schema_editor):
+        from job.configuration.configuration.job_configuration import JobConfiguration, MODE_RO, MODE_RW
+        from job.configuration.data.job_data import JobData
+        # Go through all of the job models that have job data and populate their configuration
+        Job = apps.get_model('job', 'Job')
+        ScaleFile = apps.get_model('storage', 'ScaleFile')
+        Workspace = apps.get_model('storage', 'Workspace')
+        total_count = Job.objects.all().count()
+        print 'Populating new configuration field for %s jobs' % str(total_count)
+        done_count = 0
+        for job in Job.objects.select_related('job_type').iterator():
+            if done_count % 1000 == 0:
+                percent = (done_count / total_count) * 100.00
+                print 'Completed %s of %s jobs (%f%%)' % (done_count, total_count, percent)
+            done_count += 1
+
+            # Ignore jobs that don't have their job data populated yet
+            if not job.data:
+                continue
+
+            data = JobData(job.data)
+            input_file_ids = data.get_input_file_ids()
+            input_files = ScaleFile.objects.filter(id__in=input_file_ids).select_related('workspace').iterator()
+            input_workspaces = set()
+            for input_file in input_files:
+                input_workspaces.add(input_file.workspace.name)
+
+            configuration = JobConfiguration()
+            for name in input_workspaces:
+                configuration.add_job_task_workspace(name, MODE_RO)
+            if not job.job_type.is_system:
+                for name in input_workspaces:
+                    configuration.add_pre_task_workspace(name, MODE_RO)
+                    # We add input workspaces to post task so it can perform a parse results move if requested by the
+                    # job's results manifest
+                    configuration.add_post_task_workspace(name, MODE_RW)
+                for workspace in Workspace.objects.filter(id__in=data.get_output_workspace_ids()).iterator():
```

Comment:
If we're concerned with how long this migration might take to complete, it might help to hoist the Workspace query outside all the loops. Since the number of workspaces is small, we could just query for all of them up front and cache it.